Detectors/TRD/calibration/src/CalibratorVdExB.cxx

```diff
@@ -101,13 +100,45 @@ void CalibratorVdExB::initProcessing()
   if (mInitDone) {
     return;
   }
+
+  auto& ccdbmgr = o2::ccdb::BasicCCDBManager::instance();
```

Comment:
We cannot use the CCDB manager in any DPL device running in the synchronous stage, so this has to be done differently. I think we should subscribe to the `TRD/Calib/CalVdriftExB` objects and then check the creation time of the object we get to decide whether it should be used or not. The tracklet transformer is already subscribing to it, so you could take some code from there: https://github.com/AliceO2Group/AliceO2/blob/3d9ad243b0cff6cfb3112b5d950ca01e69bd7266/Detectors/TRD/workflow/src/TRDTrackletTransformerSpec.cxx#L153

Reply:
Ok, I have updated the code a bit but I am struggling wrapping my head around how to progress further.
How can I go from dataRequest to checking the objects creation date and extracting the data in VdAndExBCalibDevice::init(...) or do I need to this in run(...)?

Reply:
Actually, even with the current state you are not really checking the time the object was created. `latest` will always give you a valid object. The default validity for one vDrift object is 1 hour. In case there was no new object created in the last hour you will get the default object which has infinite validity. So I think you can always take simply the object the framework is providing to you together with the first processed time frame.
Also note that you should not use something like ` auto currTime = std::time(nullptr) * 1000;` in the code. In case we do the processing asynchronously to the data taking that time will be wrong. Instead, you can get the creation time for the TF in question from the data header: see slide 7 of https://docs.google.com/presentation/d/1WPNixo57zAn_gM6vMCIBOHs-DZqcKAkLX0CwUgO_VYg/edit#slide=id.g10e79f44b99_0_0

Reply:
Makes sense with async processing not to take current time like this.
I am still not sure if with `pc.inputs().get<...>` I get a valid object related to that timeframe? there is something like getForTimeStamp for this, right?

Reply:
You will get the object which is valid for the current time (creation time of the first TF in flight). This can either be the default object or the last calibration output. The latter one has precedence in case both are present

Reply:
Can I check whether the object is the default or not?

Reply:
Not directly. Either by going through the values and checking that everything is at the default. Or we need to add an additional data member to the CCDB object for that. But that would mean that we have to recreate at least the default object for the CCDB. So I would simply check all values. Shouldn't hurt too much, since it needs to be done only once